### PR TITLE
Add nodes toleration to Ray pods

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?

This PR adds nodes toleration to Ray nodes

## Related issue number 

#620

### CODE REVIEW

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py [hunk 1]
```diff
@@ -121,41 +121,37 @@ def create_ray_cluster(
...
-            namespace=namespace,
-            cpu=cpus,
-            memory=memory,
-            gpu=gpus,
-            gpu_accelerator=accelerator,
-        )
+        dct["name"]=head_node_template_name
+        dct["namespace"]=namespace
+        if "tolerations" in head_node:
+            dct["tolerations"] = head_node.get("tolerations")
```
Comment (blublinsky) [inline kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py:133]: tolerations is a list of dictionaries. How do you pass it as parameter to workflow?

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py [hunk 2]
```diff
@@ -121,41 +121,37 @@ def create_ray_cluster(
...
-                name=worker_node_template_name,
-                namespace=namespace,
-                cpu=cpus,
-                memory=memory,
-                gpu=gpus,
-                gpu_accelerator=accelerator,
-            )
+            dct["name"] = worker_node_template_name
+            dct["namespace"] = namespace
+            if "tolerations" in worker_node:
```
Comment (blublinsky) [inline kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py:151]: Same as above

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py (hunk 1 above)
Comment (revit13) [reply kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py:133]: It was added to ray_head_options/ray_worker_options in the workflow as follows:
```
  "cpu": 1,
  "image": "quay.io/dataprep1/data-prep-kit/noop-ray:latest",
  "memory": 4,
  "tolerations": [
    {
      "key": "preprocessing",
      "operator": "Equal",
      "value": "dedicated",
      "effect": "NoSchedule"
    }
  ]
}
```

Comment (blublinsky) [reply kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py:133]: ok

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py (hunk 2 above)
Comment (blublinsky) [reply kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py:151]: ok

**Comment by blublinsky** [comment]
Also for some reason, you are using an old ci/cd

**Review by blublinsky — APPROVED** [review]
LGTM

**Review by roytman — APPROVED** [review]
LFTM

**Pull request merged**